src/hooks/useSearchSubmit.ts

```diff
@@ -22,8 +23,22 @@ export function useSearchSubmit(): (query: string) => void {
       // Orders, transactions, tokens, batches
       const path = pathAccordingTo(query)
       const pathPrefix = prefixNetwork ? `${prefixNetwork}/${path}` : `${path}`
-
-      query && query.length > 0 && history.push(`/${pathPrefix}/${query}`)
+      if (pathPrefix === 'address' && query.match(/[a-zA-Z0-9]+\.[a-zA-Z]+$/)?.input !== undefined) {
```

Comment:
Could you refactor the ENS regex and reuse here and in the `isAnAddressAccount` helper?

I'm thinking something like:

`isEns(query:string) => boolean`

Reply:
Also, I was testing locally and you'd want to compare 'address' against `path`, not `pathPrefix`

Reply:
Requested changes are made and added a test for the new helper.